cranelift-codegen/meta/src/srcgen.rs

```diff
@@ -151,8 +164,27 @@ impl Formatter {
                         }
                     })
                     .collect();
-                let lhs = conditions.join(" | ");
-                fmtln!(fmt, "{} => {{", lhs);
+
+                for (i, condition) in conditions.iter().enumerate() {
```

Comment:
Is there any point in caring about the output? Can't we just run rustfmt on the file if we want to read it?

Reply:
Running rustfmt on the output (from the build program itself) is a fun idea! In the meanwhile, we've tried to format code ourselves properly; using `join` here is probably clearer (I don't what why I reimplemented it from scratch myself).

Reply:
The reason I was reimplementing join by hand was because it resulted in a multi-line string, but we have the `multi_line` method designed exactly for this case, so I could use that.